org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/UriExtensionsTest.java

```diff
@@ -117,6 +119,20 @@ public void test_toUri_14() throws Exception {
 				uriExtensions.toUri("file:///c%3A/Users/dietrich/%D0%A4%D1%83%20%D0%91%D0%B0%D1%80.mydsl"));
 	}
 
+	@Test
+	public void test_toUri_15() throws Exception {
+		String uri = "something:/path/to/res%20ource";
+		String encodedUri = URLEncoder.encode(uri, StandardCharsets.UTF_8.toString());
```

Comment:
Sorry for the ping-pong here.

What is the *intent* of this test?
URLEncoder.encode(..) will encode the string as if it was a single segment of a URI.
Instead I'd assume that the intention was to have a URI with the *scheme* being `something` and three segments, right?

Reply:
The intent of the test is to show, that if there is resource where the name contains a sequence of characters that looks like an escaped sequence (the `%20`), it is handled properly, that is, the `%20` is not "unescaped". The current code has a bug here because it double-decodes. I could test this with LSP4E by creating a file with name "res%20ource" and seeing who Xtext current code takes this as "res ource" (the `%20` is replaced by ` `).